functional_tests/steps/topic_page.py

```diff
@@ -167,3 +168,124 @@ def the_time_series_item_appears_in_the_table_of_contents(context: Context) -> N
+    context_attrs = {
+        "articles": "articles",
+        "methodologies": "methodologies",
+    }
+
+    if item_type not in builder_methods:
+        raise ValueError(f"Unsupported item_type: {item_type}")
+
+    created_items = builder_methods[item_type](topic_page, items_data)
+
+    if not hasattr(context, context_attrs[item_type]):
+        setattr(context, context_attrs[item_type], {})
+    getattr(context, context_attrs[item_type]).update(created_items)
+
+
+@when('the user visits "{topic_page_title}"')
+def user_visits_topic_page(context: Context, topic_page_title: str) -> None:
+    topic_page = context.topic_pages[topic_page_title]
+    context.page.goto(f"{context.base_url}{topic_page.url}")
+
+
+@when('the user edits "{topic_page_title}"')
+def user_edits_topic_page(context: Context, topic_page_title: str) -> None:
+    topic_page = context.topic_pages[topic_page_title]
+    edit_url = reverse("wagtailadmin_pages:edit", args=[topic_page.id])
+    context.page.goto(f"{context.base_url}{edit_url}")
+
+
+@when('the user manually adds "{item_title}" in the highlighted "{item_type}" section')
+def user_manually_adds_item(context: Context, item_title: str, item_type: str) -> None:
+    button_map = {
+        "articles": ("Add topic page related article", "Choose Article page"),
+        "methodologies": ("Add topic page related methodology", "Choose Methodology page"),
+    }
+
+    if item_type not in button_map:
+        raise ValueError(f"Unsupported item_type: {item_type}")
+
+    add_button, choose_button = button_map[item_type]
+    context.page.get_by_role("button", name=add_button).click()
+    context.page.get_by_role("button", name=choose_button).click()
+    context.page.get_by_role("link", name=item_title).click()
+
+
+@then('the highlighted "{section_type}" section is visible')
+def highlighted_section_visible(context: Context, section_type: str) -> None:
+    """Check if the highlighted articles or methodologies section is visible."""
+    section_map = {
+        "articles": ("#related-articles", "Related articles"),
+        "methodologies": ("#related-methods", "Methods and quality information"),
+    }
+
+    if section_type not in section_map:
+        raise ValueError(f"Unsupported section_type: {section_type}")
+
+    selector, expected_text = section_map[section_type]
+    expect(context.page.locator(selector)).to_contain_text(expected_text)
+
+
+@then('the highlighted "{item_type}" are displayed in this order')
+def check_highlighted_items_order(context: Context, item_type: str) -> None:
+    """Check the order of highlighted articles or methodologies matches the table."""
+    expected_titles = [row[0] for row in context.table]
+    document_list = context.page.locator("ul.ons-document-list").first
+    list_items = document_list.locator("li.ons-document-list__item").all()
+
+    actual_titles = []
+    for item in list_items:
+        title_link = item.locator("h3.ons-document-list__item-title a").first
```

Comment:
Super nit-pick: the elements in your selectors are unnecessary, e.g. `ul.ons-document-list` -> `.ons-document-list`. There won't be (at least shouldn't be) a different element with this class name.